test/functional/ex_cmds/wviminfo_spec.lua

```diff
@@ -0,0 +1,54 @@
+local helpers, lfs, os =
+  require('test.functional.helpers'), require('lfs'), require('os')
+local clear, execute, eq, neq, spawn, nvim_prog, set_session =
+  helpers.clear, helpers.execute, helpers.eq, helpers.neq, helpers.spawn,
+  helpers.nvim_prog, helpers.set_session
+
+describe(':wviminfo', function()
+  local file = './foo'
+  before_each(function()
+    clear()
+    os.remove(file)
+  end)
+  after_each(function() os.remove(file) end)
+
+  it('creates a file', function()
+    -- TODO
+    -- Set up the nvim session to be able to write viminfo files.
+    local nvim2 = spawn({nvim_prog, '-u', 'NONE', '--embed'})
+    --local nvim2 = spawn({nvim_prog, '-u', 'NONE', '--embed', '--cmd', 'let hans=42' })
+    set_session(nvim2)
+    --eq(43, eval('hans'))
+
+    -- Assert that the file does not exist.
+    eq(nil, lfs.attributes(file))
+    execute('wv! '..file)
+    -- Assert that the file does exist.
+    neq(nil, lfs.attributes(file))
+  end)
+
+  it('overwrites existing files', function()
+    -- TODO
+    local nvim2 = spawn({nvim_prog, '-u', 'NONE', '--embed'})
+    set_session(nvim2)
+
+    local text = 'foo test'
+
+    local fp = io.open(file, 'w')
+    fp:write(text)
+    fp:flush()
+    fp:close()
+    -- Assert that the file already exists.
+    neq(nil, lfs.attributes(file))
+    -- Save the modification time of the file.
+    timestamp = lfs.attributes(file, 'modification')
+    -- lua does not have a sleep function, use nvim.
+    --execute('sleep 2')
+    execute('wv! '..file)
+    neq(text, io.open(file):read())
+    -- Assert that the modification time changed.
+    neq(timestamp, lfs.attributes(file, 'modification'))
+  end)
+
+  teardown(function() os.remove(file) end)
```

Comment:
This is redundant with the `after_each` block. Note that `teardown` and `setup` run before/after each `describe` group, *not* after *each test* (`it` blocks). This is kind of confusing for people used to e.g. junit. See http://olivinelabs.com/busted/ for specifics.

`before_each` and `after_each` will run before/after each `it` test.